src/stk/itk_bridge/itk_io.cpp

```diff
@@ -0,0 +1,295 @@
+#include "itk_io.h"
+#include "itk_convert.h"
+
+#include <itkImageFileReader.h>
+#include <itkImageFileWriter.h>
+#include <itkImageIOBase.h>
+
+
+/*!
+ * \brief Read a specific image type with ITK.
+ */
+template<class ImageType>
+stk::Volume read(const itk::ImageIOBase::Pointer image_io)
+{
+    auto reader = itk::ImageFileReader<ImageType>::New();
+    reader->SetImageIO(image_io);
+    reader->SetFileName(image_io->GetFileName());
+
+    try {
+        reader->Update();
+    }
+    catch (const itk::ExceptionObject& e) {
+        FATAL() << "Error while reading image '" << image_io->GetFileName() << "': "
+                << e.what();
+    }
+
+    auto image = ImageType::New();
+    image->Graft(reader->GetOutput());
+    image->SetMetaDataDictionary(reader->GetMetaDataDictionary());
```

Comment:
This is one of those moments of ITK nonsense. Apparently the meta-data dictionary needs to be passed manually from the reader to the image object... I checked the docs, but I did not find any switch in the reader class. If there is one, it is very well hidden...